### PR TITLE
Respect `transaction_ignore_urls` and `transaction_ignore_user_agents` when creating transactions in the spring webflux instrumentation

### LINKED ISSUE #2514: Propertie transaction_ignore_urls not working springboot version 2.3.6.RELEASE
Hi, i'm using a propertie transaction_ignore_urls in elasticapm.properties, for ignore urls with /actuator/*, look that happen only in springboot with reactor, the propertie transaction_ignore_user_agents too not working.

## Steps to reproduce
Im using springboot version 2.3.6.RELEASE
springCloudVersion=Hoxton.SR8
apm-server 7.9.3
apm-agent java 1.29.0

elasticapm.properties: |
enabled=true
environment=dev
service_name=gtw-common
application_packages=cloud.package.gtw.common
server_urls=http://localhost:8200
enable_log_correlation=true
enable_experimental_instrumentations=true
metrics_interval=0s
disable_metrics=system.*,jvm.*
use_path_as_transaction_name=true
transaction_ignore_urls=/actuator/prometheus,/actuator/health
disable_instrumentations=micrometer,reactor
transaction_ignore_user_agents=Prometheus*,kube-probe*

## Debug logs
[elastic-apm-agent] WARN WARNING: -Xverify:none and -noverify are not supported by the Elastic APM Java Agent. In an upcoming version, the agent will not start when these flags are set, unless the system property elastic.apm.disable_bootstrap_checks is set to true.
2022-03-15 09:41:07,009 [Attach Listener] INFO  co.elastic.apm.agent.util.JmxUtils - Found JVM-specific OperatingSystemMXBean interface: com.sun.management.OperatingSystemMXBean
2022-03-15 09:41:07,032 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - Starting Elastic APM 1.29.0 as gtw-common on Java 11.0.14 Runtime version: 11.0.14+9-Ubuntu-0ubuntu2.20.04 VM version: 11.0.14+9-Ubuntu-0ubuntu2.20.04 (Ubuntu) Linux 5.10.60.1-microsoft-standard-WSL2
2022-03-15 09:41:07,032 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - service_name: 'gtw-common' (source: Attachment configuration)
2022-03-15 09:41:07,032 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - environment: 'dev' (source: Attachment configuration)
2022-03-15 09:41:07,032 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - disable_instrumentations: 'micrometer, reactor' (source: Attachment configuration)
2022-03-15 09:41:07,032 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - enable_experimental_instrumentations: 'true' (source: Attachment configuration)
2022-03-15 09:41:07,033 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - server_urls: 'http://localhost:8200' 
2022-03-15 09:41:07,033 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - metrics_interval: '0s' (source: Attachment configuration)
2022-03-15 09:41:07,033 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - disable_metrics: 'system.*,jvm.*' (source: Attachment configuration)
2022-03-15 09:41:07,033 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - application_packages: 'cloud.lyceum.gtw.common' (source: Attachment configuration)
2022-03-15 09:41:07,033 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - enable_log_correlation: 'true' 
2022-03-15 09:41:07,033 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - transaction_ignore_urls: '/actuator/prometheus,/actuator/health' (source: Attachment configuration)
2022-03-15 09:41:07,033 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - transaction_ignore_user_agents: 'Prometheus*,kube-probe*' (source: Attachment configuration)
2022-03-15 09:41:07,033 [Attach Listener] INFO  co.elastic.apm.agent.configuration.StartupInfo - use_path_as_transaction_name: 'true' (source: Attachment configuration)
2022-03-15 09:41:08,590 [Attach Listener] INFO  co.elastic.apm.agent.impl.ElasticApmTracer - Tracer switched to RUNNING state
2022-03-15 09:41:08,644 [elastic-apm-server-healthcheck] WARN  co.elastic.apm.agent.report.ApmServerHealthChecker - Elastic APM server http://localhost:8200/ is not available (Connection refused (Connection refused))
2022-03-15 09:41:08,990 [elastic-apm-remote-config-poller] ERROR co.elastic.apm.agent.configuration.ApmServerConfigurationSource - Connection refused (Connection refused)

Ps: ignore the part  Connection refused, im using env local at this moment.
(source: Attachment configuration)
Captura de tela 2022-03-15 094429

### PR DESCRIPTION
**Pull request by tobiasstadler**
## What does this PR do?
Fixes #2514 

## Checklist
- [x] This is a bugfix
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [x] I have added tests that would fail without this fix

### CODE REVIEW

File: apm-agent-plugins/apm-spring-webflux/apm-spring-webflux-plugin/src/main/java/co/elastic/apm/agent/springwebflux/WebfluxHelper.java [hunk 1]
```diff
@@ -110,6 +112,22 @@ public static boolean isServletTransaction(ServerWebExchange exchange) {
         return Boolean.TRUE == exchange.getAttributes().get(SERVLET_TRANSACTION);
     }
 
+    public static boolean isExcluded(WebConfiguration webConfiguration, RequestPath requestPath, @Nullable String userAgentHeader) {
```
[the remark this answers is not in the document]
Comment (SylvainJuge) [reply apm-agent-plugins/apm-spring-webflux/apm-spring-webflux-plugin/src/main/java/co/elastic/apm/agent/springwebflux/WebfluxHelper.java:115]: I have just opened https://github.com/elastic/apm-agent-java/pull/2524 for that, once this is merged I'll remove the duplication.

**Review by SylvainJuge — APPROVED** [review]

**Comment by SylvainJuge** [comment]
@elasticmachine run elasticsearch-ci/docs

File: apm-agent-plugins/apm-spring-webflux/apm-spring-webflux-plugin/src/main/java/co/elastic/apm/agent/springwebflux/WebfluxHelper.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobiasstadler) [reply apm-agent-plugins/apm-spring-webflux/apm-spring-webflux-plugin/src/main/java/co/elastic/apm/agent/springwebflux/WebfluxHelper.java:115]: Thank You!

**Comment by tobiasstadler** [comment]
Thank You!

**Pull request merged**